src/averageAges.js

```diff
@@ -54,11 +80,24 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const MALE_SEX = 'm';
+  const childrenWithMothers = people.filter(({ mother, sex }) => {
+    const isMotherInList = people.find(({ name }) => name === mother);
```

Comment:
`find` doesn't return `boolean` value, so try to make up better name(that not starts with `is`)